crates/services/gas_price_service/src/v1/service.rs

```diff
@@ -303,12 +309,61 @@ where
         AtomicStorage::commit_transaction(storage_tx)?;
         let new_algo = self.algorithm_updater.algorithm();
         tracing::debug!("Updating gas price: {}", &new_algo.calculate());
-        self.shared_algo.update(new_algo);
+        self.shared_algo.update(new_algo).await;
+        let best_recorded_height = new_recorded_height.or(old_recorded_height);
+        Self::record_metrics(&metadata, gas_price, best_recorded_height);
         // Clear the buffer after committing changes
         self.da_block_costs_buffer.clear();
         Ok(())
     }
 
+    fn record_metrics(
```

Comment:
we should conditionally export metrics, see description of https://github.com/FuelLabs/fuel-core/pull/2310 for reference